### PR TITLE
Fix to prevent removal of jackd2, if installed, when installing porta…

### PR DESCRIPTION
**Pull request by WaylonR**
…udio packages.

### CODE REVIEW

File: tools/debian_buildenv.sh [hunk 1]
```diff
@@ -27,6 +27,12 @@ case "$COMMAND" in
         esac
 
         sudo apt-get update
+        if [ $(dpkg-query -W -f='${Status}' jackd2 2>/dev/null | grep -c "ok installed") -eq 1 ];
```
Comment (Be-ing) [inline tools/debian_buildenv.sh:30]: Please add a comment explaining this. It is not obvious.

Comment (WaylonR) [reply tools/debian_buildenv.sh:30]: If jackd2 is installed, as per the dpkg database, then, install libjack-jackd2-dev

**Comment by WaylonR** [comment]
I think this answers https://bugs.launchpad.net/mixxx/+bug/1464120

File: tools/debian_buildenv.sh [hunk 2]
```diff
@@ -27,6 +27,13 @@ case "$COMMAND" in
         esac
 
         sudo apt-get update
+        if [ $(dpkg-query -W -f='${Status}' jackd2 2>/dev/null | grep -c "ok installed") -eq 1 ]; 
+		# If jackd2 is installed as per dpkg database, install libjack-jackd2-dev.
```
Comment (Be-ing) [inline tools/debian_buildenv.sh:31]: This comment still doesn't tell me why this code is here. What would be the problem if it wasn't there?

Comment (Be-ing) [reply tools/debian_buildenv.sh:31]: Comments should explain *why* rather than *how*

Comment (Be-ing) [reply tools/debian_buildenv.sh:31]: The comment is better now. It's pretty unusual to put a comment in the middle of an `if...then` clause. Please move it above the `if`

**Comment by Be-ing** [comment]
pre-commit shellcheck failed:
```
shellcheck...............................................................Failed
- hook id: shellcheck
- exit code: 1

In tools/debian_buildenv.sh line 35:
        if [ $(dpkg-query -W -f='${Status}' jackd2 2>/dev/null | grep -c "ok installed") -eq 1 ];
             ^-- SC2046: Quote this to prevent word splitting.
```

File: tools/debian_buildenv.sh [hunk 3]
```diff
@@ -27,6 +27,17 @@ case "$COMMAND" in
         esac
 
         sudo apt-get update
+
+	# If jackd2 is installed as per dpkg database, install libjack-jackd2-dev.
+        # This avoids a package deadlock, resulting in jackd2 being removed, and jackd1 being installed,
+        # to satisfy portaudio19-dev's need for a jackd dev package. In short, portaudio19-dev needs a
+        # jackd dev library, so let's give it one..
+        if [ "$(dpkg-query -W -f='${Status}' jackd2 2>/dev/null | grep -c "ok installed")" -eq 1 ];
```
Comment (Be-ing) [inline tools/debian_buildenv.sh:35]:
```suggestion
        if [ "$(dpkg-query -W -f='${Status}' jackd2 2>/dev/null | grep -c "ok installed")" -eq 1 ];
```
extra space at the end of the line

File: tools/debian_buildenv.sh [hunk 4]
```diff
@@ -27,6 +27,17 @@ case "$COMMAND" in
         esac
 
         sudo apt-get update
+
+	# If jackd2 is installed as per dpkg database, install libjack-jackd2-dev.
```
Comment (Be-ing) [inline tools/debian_buildenv.sh:31]:
```suggestion
	  # If jackd2 is installed as per dpkg database, install libjack-jackd2-dev.
```
fix indentation

**Comment by Be-ing** [comment]
Thank you!

Could you sign the [Contributor License Agreement](https://docs.google.com/forms/d/e/1FAIpQLScC9QG327sjLL0eWftmfGUasxFFLxg6LCXJ2xHDYRzFIRqyiw/viewform?formkey=dEpYN2NkVEFnWWQzbkFfM0ZYYUZ5X2c6MQ)? Sorry, I should have asked before pressing merge.

**Comment by WaylonR** [comment]
Signed.

**Comment by Be-ing** [comment]
thanks

**Pull request merged**